File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-2.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func inorderTraversal(root *TreeNode) []int {
	arr := []int{}
	stack := []*TreeNode{}
	cur := root 
	for cur != nil || len(stack) > 0 {
		for cur != nil {
			stack = append(stack, cur)
			cur = cur.Left
		}

		// top of stack
		cur = stack[len(stack) - 1]
		// pop stack
		stack = stack[:len(stack)-1]
		arr = append(arr, cur)
		cur = cur.Right
	}

	return arr
}

File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:   
    def serialize(self, node: Optional[TreeNode]):
      if not node:
        return "#"
      left = self.serialize(node.left)
      right = self.serialize(node.right)
      return f":{node.val}:{left},{right}"
      
    def isSubtree(self, root: Optional[TreeNode], subRoot: Optional[TreeNode]) -> bool:
      root_sl = self.serialize(root)
      sub_sl = self.serialize(subRoot)
      return sub_sl in root_sl